mcs/class/corlib/Test/System/ArrayTest.cs

```diff
@@ -1002,8 +1002,7 @@ public void CreateInstanceVoid ()
 		int[] myLengthArray = new int[1] { 3 };
 		int[] myBoundArray = new int[1] { Int32.MinValue };
 		Array myExtremeArray=Array.CreateInstance ( typeof(String), myLengthArray, myBoundArray );
-		Assert.AreEqual (Int32.MaxValue, ((IList)myExtremeArray).IndexOf (42), "AD04");
-
+		Assert.AreEqual (-1, ((IList)myExtremeArray).IndexOf (42), "AD04");
```

Comment:
This special case is documented on MSDN so it's not a bug per-se: https://msdn.microsoft.com/en-us/library/7eddebat(v=vs.110).aspx#Anchor_2

> Because most arrays have a lower bound of zero, this method generally returns –1 if value isn’t found. In the rare case that the lower bound of the array is equal to Int32.MinValue(0x80000000) and value isn’t found, this method returns Int32.MaxValue (0x7FFFFFFF).